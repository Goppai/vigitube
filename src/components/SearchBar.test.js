import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { SearchBar } from './'

describe('<SearchBar/>', () => {
  const search = jest.fn()
  const wrapper = shallow(<SearchBar search={search}/>)

  it('renders', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders input', () => {
    expect(wrapper.find('.search-input').length).toBe(1)
  })

  it('handles input change', () => {
    const input = wrapper.find('.search-input')
    input.simulate('change', { target: { value: 'testing' } })
    expect(search).toHaveBeenCalled()
  })
})
