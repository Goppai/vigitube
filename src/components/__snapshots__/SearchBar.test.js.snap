// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> renders 1`] = `
<Col
  className="search-bar"
  xs={24}
>
  <a
    className="vigitube"
  >
    <span
      className="vigitube-part-one"
    >
      Vigi
    </span>
    <span
      className="vigitube-part-two"
    >
      Tube
    </span>
  </a>
  <a
    className="github-repo-link"
    href=""
    rel="noopener noreferrer"
    target="_blank"
  >
    <Icon
      type="github"
    />
  </a>
  <Debounce
    handler="onChange"
    handlers={Array []}
    time="500"
  >
    <Search
      className="search-input"
      enterButton={true}
      inputPrefixCls="ant-input"
      onChange={[Function]}
      onSearch={[Function]}
      placeholder="Search"
      prefixCls="ant-input-search"
      size="large"
    />
  </Debounce>
</Col>
`;
